h/api/models/test/elastic_test.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+
+import pytest
+from mock import patch
+from mock import PropertyMock
+
+from h.api.models.elastic import Annotation
+
+
+def test_uri():
+    assert Annotation(uri="http://foo.com").uri == "http://foo.com"
+
+
+def test_uri_with_no_uri():
+    assert Annotation().uri == ""
+
+
+def test_uri_when_uri_is_not_a_string():
+    for uri in (True, None, 23, 23.7, {"foo": False}, [1, 2, 3]):
```

Comment:
No. This is all to deal with the fact that we basically accept anything POSTed to `/api/annotations`. [As yet](https://github.com/hypothesis/h/blob/move-es-config-into-search/h/api/schemas.py#L55) there is no validation that the `uri` field or the `target.source` is a string.